### ISSUE TITLE
mkdir: fails to create directory in -p mode if path ends in /.

### ISSUE THREAD

**Issue by jfinkels** [uutils/coreutils#3213]
`mkdir -p` fails to create a directory when the path ends in `/.`, but it should work.

GNU mkdir:
```
$ mkdir -p dir/.  # succeeds, and dir/ is created
```

uutils mkdir:
```
$ ./target/debug/mkdir -p dir/.
./target/debug/mkdir: cannot create directory 'dir/.': No such file or directory
```

**Comment by DevSabb** [comment]
The command is pretty much outsourced to rust std lib. So, I guess, the behavior of the stdlib has to be changed to match mkdir spec.
https://doc.rust-lang.org/std/fs/fn.create_dir_all.html

**Comment by jfinkels** [comment]
Fixed by pull request #3311.
